Userland/Libraries/LibGUI/TextDocument.cpp

```diff
@@ -963,10 +963,11 @@ void InsertTextCommand::undo()
     m_document.set_all_cursors(m_range.start());
 }
 
-RemoveTextCommand::RemoveTextCommand(TextDocument& document, DeprecatedString const& text, TextRange const& range)
+RemoveTextCommand::RemoveTextCommand(TextDocument& document, DeprecatedString const& text, TextRange const& range, RemoveDirection direction)
     : TextDocumentUndoCommand(document)
     , m_text(text)
     , m_range(range)
+    , m_direction(direction)
```

Comment:
Instead of passing a new `RemoveDirection` enum, wouldn't it be simpler to pass in the cursor position to restore?

Reply:
Yeah, using enum brings unnecessary branches in `RemoveTextCommand::undo()`, I've changed `RemoveTextCommand` to use `m_original_cursor_position` to restore the cursor.